res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -1887,6 +1843,54 @@ NumarkMixtrack3.BeatKnob = function(channel, control, value, status, group) {
 
         engine.setValue(deck.group, 'beatjump_size', jumpSize);
     }
+
+    // tap to move beat grid
+    if (deck.TapDown) {
+        if (increase) {
+            engine.setValue(deck.group, 'beats_translate_later', true);
+        } else {
+            engine.setValue(deck.group, 'beats_translate_earlier', true);
+        }
+    }
+
+    // pad to change sampler gains
+    if (deck.PADMode) {
+        var numSamplers = engine.getValue('[Master]', 'num_samplers');
+        var samplersPerRack = 8;
+        var samplersPerSide = 4;
+        var samplersToToggle = [];
+        var startingSampler = (deck.decknum < 3) ? 1 : samplersPerSide + 1;
+
+        // divvy up samplers that are associated with left and right decks
+        for (var i = startingSampler, n = numSamplers; i <= n; i = i + samplersPerRack) {
+            for (var j = 0; j < samplersPerSide; j++) {
+                samplersToToggle.push(i + j);
+            }
+        }
+
+        // adjust pre-gain for entire left/right ggroup
+        for (var i = 0; i <= samplersToToggle.length; i++) {
+            var samplerGroup = '[Sampler' + samplersToToggle[i] + ']'
+            var gainValue = engine.getValue(samplerGroup, 'pregain');
+            var gainMultiplier = 3;
+            var maxGain = 4;
```

Comment:
I'm a little unclear on the difference between `getValue` and `getParameter` tbh. Basically parameters perform scaling and bound checking between 0 and 1?

> “Parameter” values range from 0 to 1. If you set a parameter to 1 then the value of the control becomes the maximum value. If you set the parameter to 0 then the value of the control becomes the minimum value.

So basically `setParameter` and `setValue` are almost always preferred?